server/node-server.js

```diff
@@ -34,6 +36,9 @@ passport.use(new FacebookStrategy({
 // Facebook will redirect the user back to the application at
 //     /auth/facebook/callback
 app.get('/auth/facebook', passport.authenticate('facebook'));
+app.get('/me', checkAuthorizationToken(), authenticate(), (request, response) => {
+  response.json(request.user);
+});
```

Comment:
So yeah, anyone who hits an authenticated route needs to provide an `Authorization: bearer ${token}` header. This token is basically the one we return to the front-end when we initially ask the user to authorize our application. Using that token, we'll hit Facebook's graph API to fetch the relevant user. If the user was successfully fetched, the request will be decorated with a `user` property with the fetched user. Otherwise, if any sort of error occurs, then we'll reject the request with a 401 error.

So, to require authentication for any route, we simply need to do the following:

```JavaScript
import authenticate from './middleware/authenticate/facebook';

app.get('/share-links', authenticate(), (request, response) => {
  // user is now authenticated and available on request.user
  response.json(request.user);
});
```